// @paddle/paddle-node-sdk/dist/esm/resources/prices/index.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { BaseResource, PathParameters, QueryParameters } from '../../internal/base/index.js';
import { Price, PriceCollection } from '../../entities/index.js';
const PricePaths = {
    list: '/prices',
    create: '/prices',
    get: '/prices/{price_id}',
    update: '/prices/{price_id}',
};
export * from './operations/index.js';
export class PricesResource extends BaseResource {
    list(queryParams) {
        const queryParameters = new QueryParameters(queryParams);
        return new PriceCollection(this.client, PricePaths.list + queryParameters.toQueryString());
    }
    create(createPriceParameters) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.client.post(PricePaths.create, createPriceParameters);
            const data = this.handleResponse(response);
            return new Price(data);
        });
    }
    get(priceId, queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            const queryParameters = new QueryParameters(queryParams);
            const urlWithPathParams = new PathParameters(PricePaths.get, {
                price_id: priceId,
            }).deriveUrl();
            const response = yield this.client.get(urlWithPathParams, queryParameters);
            const data = this.handleResponse(response);
            return new Price(data);
        });
    }
    update(priceId, updatePrice) {
        return __awaiter(this, void 0, void 0, function* () {
            const urlWithPathParams = new PathParameters(PricePaths.update, {
                price_id: priceId,
            }).deriveUrl();
            const response = yield this.client.patch(urlWithPathParams, updatePrice);
            const data = this.handleResponse(response);
            return new Price(data);
        });
    }
    archive(priceId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.update(priceId, { status: 'archived' });
        });
    }
}
