// @paddle/paddle-node-sdk/dist/esm/resources/discounts/index.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { BaseResource, PathParameters, QueryParameters } from '../../internal/base/index.js';
import { Discount, DiscountCollection } from '../../entities/index.js';
const DiscountPaths = {
    list: '/discounts',
    create: '/discounts',
    get: '/discounts/{discount_id}',
    update: '/discounts/{discount_id}',
};
export * from './operations/index.js';
export class DiscountsResource extends BaseResource {
    list(queryParams) {
        const queryParameters = new QueryParameters(queryParams);
        return new DiscountCollection(this.client, DiscountPaths.list + queryParameters.toQueryString());
    }
    create(createDiscountParameters) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.client.post(DiscountPaths.create, createDiscountParameters);
            const data = this.handleResponse(response);
            return new Discount(data);
        });
    }
    get(discountId) {
        return __awaiter(this, void 0, void 0, function* () {
            const urlWithPathParams = new PathParameters(DiscountPaths.get, {
                discount_id: discountId,
            }).deriveUrl();
            const response = yield this.client.get(urlWithPathParams);
            const data = this.handleResponse(response);
            return new Discount(data);
        });
    }
    update(discountId, updateDiscount) {
        return __awaiter(this, void 0, void 0, function* () {
            const urlWithPathParams = new PathParameters(DiscountPaths.update, {
                discount_id: discountId,
            }).deriveUrl();
            const response = yield this.client.patch(urlWithPathParams, updateDiscount);
            const data = this.handleResponse(response);
            return new Discount(data);
        });
    }
    archive(discountId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.update(discountId, { status: 'archived' });
        });
    }
}
