// @paddle/paddle-node-sdk/dist/cjs/resources/notification-settings/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationSettingsResource = void 0;
const index_js_1 = require("../../internal/base/index.js");
const index_js_2 = require("../../entities/index.js");
const NotificationSettingsPaths = {
    list: '/notification-settings',
    create: '/notification-settings',
    get: '/notification-settings/{notification_setting_id}',
    update: '/notification-settings/{notification_setting_id}',
    delete: '/notification-settings/{notification_setting_id}',
};
__exportStar(require("./operations/index.js"), exports);
class NotificationSettingsResource extends index_js_1.BaseResource {
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            const queryParameters = new index_js_1.QueryParameters(queryParams);
            const response = yield this.client.get(NotificationSettingsPaths.list + queryParameters.toQueryString());
            const data = this.handleResponse(response);
            return data.map((notificationSetting) => new index_js_2.NotificationSettings(notificationSetting));
        });
    }
    create(createNotificationSettings) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.client.post(NotificationSettingsPaths.create, createNotificationSettings);
            const data = this.handleResponse(response);
            return new index_js_2.NotificationSettings(data);
        });
    }
    get(notificationId) {
        return __awaiter(this, void 0, void 0, function* () {
            const urlWithPathParams = new index_js_1.PathParameters(NotificationSettingsPaths.get, {
                notification_setting_id: notificationId,
            }).deriveUrl();
            const response = yield this.client.get(urlWithPathParams);
            const data = this.handleResponse(response);
            return new index_js_2.NotificationSettings(data);
        });
    }
    update(notificationId, updateNotificationSettings) {
        return __awaiter(this, void 0, void 0, function* () {
            const urlWithPathParams = new index_js_1.PathParameters(NotificationSettingsPaths.update, {
                notification_setting_id: notificationId,
            }).deriveUrl();
            const response = yield this.client.patch(urlWithPathParams, updateNotificationSettings);
            const data = this.handleResponse(response);
            return new index_js_2.NotificationSettings(data);
        });
    }
    delete(notificationId) {
        return __awaiter(this, void 0, void 0, function* () {
            const urlWithPathParams = new index_js_1.PathParameters(NotificationSettingsPaths.update, {
                notification_setting_id: notificationId,
            }).deriveUrl();
            const response = yield this.client.delete(urlWithPathParams);
            if (response) {
                this.handleResponse(response);
            }
        });
    }
}
exports.NotificationSettingsResource = NotificationSettingsResource;
